consistence/nsq_lookupd_etcd.go

```diff
@@ -373,15 +379,15 @@ func (self *NsqLookupdEtcdMgr) watchTopics() {
 			}
 		}
 		coordLog.Debugf("topic changed.")
-		atomic.StoreInt32(&self.ifTopicChanged, 1)
+		self.touchTopicChanged()
 	}
 }
 
 func (self *NsqLookupdEtcdMgr) scanTopics() ([]TopicPartitionMetaInfo, error) {
 	atomic.StoreInt32(&self.ifTopicChanged, 0)
 	rsp, err := self.client.Get(self.topicRoot, true, true)
 	if err != nil {
-		atomic.StoreInt32(&self.ifTopicChanged, 1)
+		self.touchTopicChanged()
 		if client.IsKeyNotFound(err) {
```

Comment:
```

	self.tmisMutex.Lock()
 	self.topicMetaInfos = topicMetaInfos
 	self.tmisMutex.Unlock()
```
We can save the scanned `topicMetaMap` to the `self.topicMetaMap`. Then we got the cached meta map. If any `ifTopicChanged` we treat as cache invalid and got meta directly from etcd